File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> m;
        for(int i=0;i<nums.size();i++){
            m[nums[i]] = i;
        }

        for(int i=0;i<nums.size();i++){
            int diff = target - nums[i];
            auto it = m.find(diff);
            //cout << it;
            if(it != m.end() && i!=m[diff]){
                return {i,m[diff]};
            }
        }

        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(0,len(nums)):
            d[nums[i]] = d.get(nums[i],[])
            d[nums[i]].append(i)
        
        for i in range(0,len(nums)):
            to_find = target - nums[i]
            if to_find in d and d[to_find][0] != i:
                if to_find == nums[i] and len(d[to_find]) > 1:
                    return [d[to_find][0],d[to_find][1]]
                else:
                    return [i,d[to_find][0]]
